b2b_personalization/justfile

```
@@ -132,9 +140,151 @@ migration tenant env=aws_local_env batch_num="100" path_to_migration="~/git/xayn
+# Warning: Can only be moved from one env to the other, can not change the tenant name. 
+tenant-backup-es tenant dir env=aws_local_env:
+    #!/usr/bin/env bash
+    set -e
+    PROFILE=$({{just_executable()}} _env_to_profile {{env}}) || { ret=$? ; echo $PROFILE ; exit $ret; }
+    REGION=$({{just_executable()}} _read_region $PROFILE)
+
+    mkdir -p dir || true
+    # Get credentials
+    elasticsearch_url=$({{just_executable()}} _read_property "/elasticsearch/elasticsearch-{{env}}/url" $PROFILE $REGION)
+    elasticsearch_user=$({{just_executable()}} _read_property "/elasticsearch/elasticsearch-{{env}}/user" $PROFILE $REGION)
+    elasticsearch_pass=$({{just_executable()}} _read_property "/elasticsearch/elasticsearch-{{env}}/password" $PROFILE $REGION)
+    url="$(echo $elasticsearch_url | sed -e "s/https:\/\//https:\/\/$elasticsearch_user:$elasticsearch_pass@/"):9243/{{tenant}}"
+    
+    # Connect to bastion host
+    # port 5432
+    {{just_executable()}} connect-es {{env}} &
+    PID=$!
+    sleep 5
+    
+    elasticdump --input=$url --output="{{dir}}/mapping.json" --type=mapping --debug
+    elasticdump --input=$url --output="{{dir}}/alias.json" --type=alias --debug
+    elasticdump --input=$url --output="{{dir}}/data.json" --type=data --limit 1000 --esCompress
+
+    sleep 5
+    PIDS=$({{just_executable()}} _child_pids $PID)
+    kill -9 $PIDS $PID
+
+# Restore a elastic index, requiers to install `npm install elasticdump -g`
+# HOW TO DUPLICATE TENANTS FROM ENV A -> B (i.e. prod -> dev)
+# run backup (es and pg)
+# run restore (es and pg)
+# create the new tenant in the target infrastructure.
+# Warning: Can only be moved from one env to the other, can not change the tenant name. 
+tenant-restore-es tenant dir env=aws_local_env:
+    #!/usr/bin/env bash
+    set -e
+    PROFILE=$({{just_executable()}} _env_to_profile {{env}}) || { ret=$? ; echo $PROFILE ; exit $ret; }
+    REGION=$({{just_executable()}} _read_region $PROFILE)
+
+    # Get credentials
+    elasticsearch_url=$({{just_executable()}} _read_property "/elasticsearch/elasticsearch-{{env}}/url" $PROFILE $REGION)
+    elasticsearch_user=$({{just_executable()}} _read_property "/elasticsearch/elasticsearch-{{env}}/user" $PROFILE $REGION)
+    elasticsearch_pass=$({{just_executable()}} _read_property "/elasticsearch/elasticsearch-{{env}}/password" $PROFILE $REGION)
+    urlWithoutTenant="$(echo $elasticsearch_url | sed -e "s/https:\/\//https:\/\/$elasticsearch_user:$elasticsearch_pass@/"):9243"
+    
+    
+    # Connect to bastion host
+    # port 5432
+    {{just_executable()}} connect-es {{env}} &
+    PID=$!
+    sleep 5
+    
+
+    jq '. | to_entries[0].value' < {{dir}}/mapping.json > {{dir}}/mapping_new.json
+    jq 'fromjson | . | to_entries[0].value' < {{dir}}/alias.json > {{dir}}/alias_new.json
+    curl --header 'Content-Type: application/json'  --data @{{dir}}/mapping_new.json --request PUT  "$urlWithoutTenant/{{tenant}}"
```

:art: You could use `--json` instead of `--header` + `--data`.

(:art: I personally like to use `--url` allowing me to writ things in the order they tend to appear in in other places (method -> url -> body):

```suggestion
    curl --request PUT --url "$urlWithoutTenant/{{tenant}}" --json "@{{dir}}/mapping_new.json"
```
)